File: INF-143/contest_5/E/main.cpp
#include<bits/stdc++.h>

using namespace std;

long long int f(long long int n)
{
	return n * (n + 1) / 2;
}

int main()
{
	double x;

	cin >> x;

	double r1 = (-1 + sqrt(1 + (8 * (x - 1)))) / 2;
	long long int r2 = (-1 + sqrt(1 + (8 * (x - 1)))) / 2;

	if(r1 != r2)
	{
		long long int low = 1, high = 2 * x, mid;

		while(high - low > 1)
		{
			mid = (high + low) / 2;
			if(mid <= (2 * ((long long int)x - 1)) / (mid + 1))
				low = mid;
			else
				high = mid;
		}
		cout << to_string((long long int)x - f(low)) << "\n";
	}
	else
		cout << "1\n";	
	return 0;
}

File: INF-143/contest_5/E/main.py
import math as m
def f(n):
	return n * (n + 1) / 2

x = 1000000000000
r1 = (-1 + m.sqrt(1 + (8 * (x - 1)))) / 2
r2 = int((-1 + m.sqrt(1 + (8 * (x - 1)))) / 2)

if r1 != r2:
	low = 1
	high = 2 * 1000000000000

	while(high - low > 1):
		mid = (high + low) // 2;
		if f(mid) + 1 <= x:
			low = mid;
		else:
			high = mid;
	print(x - f(low))
else:
	print("1")
